quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMappers.java

```diff
@@ -63,6 +62,7 @@ private PropertyMappers(){}
         MAPPERS.addAll(ExportPropertyMappers.getMappers());
         MAPPERS.addAll(ImportPropertyMappers.getMappers());
         MAPPERS.addAll(TruststorePropertyMappers.getMappers());
+        sanitizeDisabledMappers();
```

Comment:
@vmuzikar I'm afraid it's not gonna work. As for the proper determination of disabled mappers, the config sources need to be initialized. AFAIK, when this is done before the cli.parseArgs[1], the `--config-file` property will not work as you'd need to reinit particular config sources after evaluating that. That was the main reason to delay the sanitization. If I remember correctly, it had some impact also on the Profile initialization, but not sure. But I believe it could be somehow "workarounded".

[1] https://github.com/keycloak/keycloak/pull/26345/files#diff-5e7526010bd1f3d3c2735d6b5d600aaf3457a83eb1699ef7e10c91e2257ae6d2L113

Reply:
Thanks for the pointer! You are correct, I implemented it in a different way.